src/openms/source/DATASTRUCTURES/ParamValue.cpp

```diff
@@ -0,0 +1,876 @@
+      break;
+      case ParamValue::DOUBLE_VALUE:
+          return a.data_.dou_ < b.data_.dou_;
+      break;
+      }
+    }
+    return false;
+  }
+
+  bool operator>(const ParamValue& a, const  ParamValue& b)
+  {
+    if (a.value_type_ == b.value_type_)
+    {
+      switch (a.value_type_)
+      {
+      case ParamValue::EMPTY_VALUE:
+          return false;
+      break;
+      case ParamValue::STRING_VALUE:
+          return *(a.data_.str_) > *(b.data_.str_);
+      break;
+      case ParamValue::STRING_LIST:
+          return a.data_.str_list_->size() > b.data_.str_list_->size();
+      break;
+      case ParamValue::INT_LIST:
+          return a.data_.int_list_->size() > b.data_.int_list_->size();
+      break;
+      case ParamValue::DOUBLE_LIST:
+          return a.data_.dou_list_->size() > b.data_.dou_list_->size();
+      break;
+      case ParamValue::INT_VALUE:
+          return a.data_.ssize_ > b.data_.ssize_;
+      break;
+      case ParamValue::DOUBLE_VALUE:
+          return a.data_.dou_ > b.data_.dou_;
+      break;
+      }
+    }
+    return false;
+  }
+
+  bool operator!=(const ParamValue& a, const ParamValue& b)
+  {
+    return !(a == b);
+  }
+
+  // ----------------- Output operator ----------------------
+
+  /// for doubles or lists of doubles, you get full precision. Use ParamValue::toString(false) if you only need low precision
+  std::ostream& operator<<(std::ostream& os, const ParamValue& p)
+  {
+    switch (p.value_type_)
+    {
+    case ParamValue::STRING_VALUE:
+        os << *(p.data_.str_);
+    break;
+    case ParamValue::STRING_LIST:
+        os << "[";
+        if (!p.data_.str_list_->empty())
+        {
+          for (auto it = p.data_.str_list_->begin(), end = p.data_.str_list_->end() - 1; it != end; ++it) 
+          {
+            os << *it << ", ";
+          }
+          os << p.data_.str_list_->back();
+        }
+        os << "]";
+    break;
+    case ParamValue::INT_LIST:
+        os << "[";
+        if (!p.data_.int_list_->empty())
+        {
+          for (auto it = p.data_.int_list_->begin(), end = p.data_.int_list_->end() - 1; it != end; ++it) 
+          {
+            os << *it << ", ";
+          }
+          os << p.data_.int_list_->back();
+        }
+        os << "]";
+    break;
+    case ParamValue::DOUBLE_LIST:
+        os << "[";
+        if (!p.data_.dou_list_->empty()) 
+        {
+          for (auto it = p.data_.dou_list_->begin(), end = p.data_.dou_list_->end() - 1; it != end; ++it) 
+          {
+            os << *it << ", ";
+          }
+          os << p.data_.dou_list_->back();
+        }
+        os << "]";
+    break;
+    case ParamValue::INT_VALUE:
+        os << p.data_.ssize_;
+    break;
+    case ParamValue::DOUBLE_VALUE:
+        os << p.data_.dou_;
+    break;
+    case ParamValue::EMPTY_VALUE:
+    break;
+    }
+    return os;
+  }
+
+    std::string ParamValue::doubleToString(double value, bool full_precision) 
```

Comment:
Hmm not sure if this is the right place here.
There is also more efficient code in StringConversions.
Indentation looks off.

Reply:
StringConversions is a non standard dependency and  can not be used here.
All the stl functions for string conversions I looked at can not replicate the required conversions (for example they won't remove trailing zeros or switch to scientific notation at a certain value).

Reply:
I think Ruben had to come up with some custom code to get the same string format

Reply:
Ok